packages/package-json-lint/src/rules/require-no-unnecessary-dependency.js

```diff
@@ -0,0 +1,124 @@
+const devDependencySet = [
+  '@babel/cli',
+  '@babel/core',
+  '@babel/helpers',
+  '@babel/plugin-proposal-object-rest-spread',
+  '@babel/plugin-syntax-async-generators',
+  '@babel/plugin-transform-object-assign',
+  '@babel/plugin-transform-regenerator',
+  '@babel/plugin-transform-runtime',
+  '@babel/preset-env',
+  '@babel/preset-react',
+  '@babel/register',
+  '@cerner/browserslist-config-terra',
+  '@cerner/eslint-config-terra',
+  '@cerner/jest-config-terra',
+  '@cerner/stylelint-config-terra',
+  '@cerner/terra-aggregate-translations',
+  '@cerner/terra-application',
+  '@cerner/terra-cli',
+  '@cerner/terra-dev-site',
+  '@cerner/terra-functional-testing',
+  '@cerner/terra-open-source-scripts',
+  '@cerner/webpack-config-terra',
+  '@octokit/core',
+  'bufferutil',
+  'chai',
+  'check-installed-dependencies',
+  'commander',
+  'core-js',
+  'csvtojson',
+  'danger',
+  'enzyme',
+  'enzyme-adapter-react-16',
+  'enzyme-to-json',
+  'eslint',
+  'gh-pages',
+  'glob',
+  'html-webpack-plugin',
+  'jest',
+  'jsdom',
+  'ky',
+  'lerna',
+  'link-parent-bin',
+  'lodash.groupby',
+  'lodash.map',
+  'lodash.mapkeys',
+  'lodash.mapvalues',
+  'memory-fs',
+  'mocha',
+  'nock',
+  'np',
+  'one-cerner-style-icons',
+  'postcss',
+  'react-test-renderer',
+  'regenerator-runtime',
+  'shelljs',
+  'sinon',
+  'strip-ansi',
+  'stylelint',
+  'svgo',
+  'terra-application',
+  'terra-button',
+  'terra-disclosure-manager',
```

Comment:
What is the reason that these terra components, `lodash`, `commander`, etc must be a dev dependency? I would think many consumers are currently have these as regular dependencies.

Reply:
I had run a script to get dev and peer dependencies from the mono-repo's. And in that some packages have these as both dependencies and devDependencies. I will go through them once and remove the unneeded dependencies from the set.